service/cloudconfig/master_extension.go

```diff
@@ -53,8 +53,100 @@ func (me *masterExtension) Units() ([]k8scloudconfig.UnitAsset, error) {
 		return nil, microerror.Maskf(err, "renderGetKeyVaultSecretsUnit")
 	}
 
+	// Unit to format disk for etcd.
+	formatEtcdUnit := k8scloudconfig.UnitAsset{
+		Metadata: k8scloudconfig.UnitMetadata{
+			AssetContent: "",
+			Name:         "format-etcd-disk.service",
+			Enable:       true,
+			Command:      "start",
+		},
+		Content: []string{`[Unit]
+      Description=Formats the disk drive
+      Requires=dev-sdc.device
+      After=dev-sdc.device
+      [Service]
+      Type=oneshot
+      RemainAfterExit=yes
+      Environment="LABEL=etcd"
+      Environment="DEV=/dev/sdc"
+      # Do not wipe the disk if it's already being used, so the etcd data is persistent across reboot.
+      ExecStart=-/bin/bash -c "if ! findfs LABEL=$LABEL > /tmp/label.$LABEL; then wipefs -a -f $DEV && mkfs.ext4 -T news -F -L $LABEL $DEV && echo wiped; fi"
+`,
+		},
+	}
+
+	// Unit to mount etcd disk.
+	mountEtcdUnit := k8scloudconfig.UnitAsset{
+		Metadata: k8scloudconfig.UnitMetadata{
+			AssetContent: "",
+			Name:         "etc-kubernetes-data-etcd.mount",
+			Enable:       true,
+			Command:      "start",
+		},
+		Content: []string{`[Unit]
+      Description=Mounts disk to /etc/kubernetes/data/etcd
+      Requires=format-etcd-disk.service
+      After=format-etcd-disk.service
+      Before=etcd3.service
+      [Mount]
+      What=/dev/sdc
+      Where=/etc/kubernetes/data/etcd
+      Type=ext4
+`,
+		},
+	}
+
+	// Unit to format docker disk.
+	formatDockerUnit := k8scloudconfig.UnitAsset{
+		Metadata: k8scloudconfig.UnitMetadata{
+			AssetContent: "",
+			Name:         "format-docker-disk.service",
+			Enable:       true,
+			Command:      "start",
+		},
+		Content: []string{`[Unit]
+      Description=Formats the disk drive
+      Requires=dev-sdd.device
+      After=dev-sdd.device
+      [Service]
+      Type=oneshot
+      RemainAfterExit=yes
+      Environment="LABEL=docker"
+      Environment="DEV=/dev/sdd"
+      # Do not wipe the disk if it's already being used.
+      ExecStart=-/bin/bash -c "if ! findfs LABEL=$LABEL > /tmp/label.$LABEL; then wipefs -a -f $DEV && mkfs.xfs -L $LABEL $DEV && echo formatted; fi"
+`,
+		},
+	}
+
+	// Unit to mount docker disk.
+	mountDockerUnit := k8scloudconfig.UnitAsset{
+		Metadata: k8scloudconfig.UnitMetadata{
+			AssetContent: "",
+			Name:         "var-lib-docker.mount",
+			Enable:       true,
+			Command:      "start",
+		},
+		Content: []string{`[Unit]
+      Description=Mounts disk to /var/lib/docker
+      Requires=format-docker-disk.service
+      After=format-docker-disk.service
+      Before=docker.service
+      [Mount]
+      What=/dev/sdd
+      Where=/var/lib/docker
+      Type=xfs
+`,
+		},
+	}
```

Comment:
I understand you wanted to avoid `render*` functions dance, but let's please do it for the sake of clarity.
Example is `renderGetKeyVaultSecretsUnit`.